src/components/trends/cbg/CBGSliceAnimated.js

```diff
@@ -0,0 +1,244 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2017, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+import _ from 'lodash';
+import cx from 'classnames';
+import React, { Component, PropTypes } from 'react';
+import { TransitionMotion, spring } from 'react-motion';
+
+import { classifyBgValue } from '../../../utils/bloodglucose';
+
+import styles from './CBGSliceAnimated.css';
+
+class CBGSliceAnimated extends Component {
+  static defaultProps = {
+    medianHeight: 10,
+    sliceWidth: 16,
+  };
+
+  static propTypes = {
+    bgBounds: PropTypes.shape({
+      veryHighThreshold: PropTypes.number.isRequired,
+      targetUpperBound: PropTypes.number.isRequired,
+      targetLowerBound: PropTypes.number.isRequired,
+      veryLowThreshold: PropTypes.number.isRequired,
+    }).isRequired,
+    datum: PropTypes.shape({
+      firstQuartile: PropTypes.number,
+      id: PropTypes.string.isRequired,
+      max: PropTypes.number,
+      median: PropTypes.number,
+      min: PropTypes.number,
+      msFrom: PropTypes.number.isRequired,
+      msTo: PropTypes.number.isRequired,
+      msX: PropTypes.number.isRequired,
+      ninetiethQuantile: PropTypes.number,
+      tenthQuantile: PropTypes.number,
+      thirdQuartile: PropTypes.number,
+    }).isRequired,
+    displayFlags: PropTypes.shape({
+      cbg100Enabled: PropTypes.bool.isRequired,
+      cbg80Enabled: PropTypes.bool.isRequired,
+      cbg50Enabled: PropTypes.bool.isRequired,
+      cbgMedianEnabled: PropTypes.bool.isRequired,
+    }).isRequired,
+    focusSlice: PropTypes.func.isRequired,
+    isFocused: PropTypes.bool.isRequired,
+    medianHeight: PropTypes.number.isRequired,
+    sliceWidth: PropTypes.number.isRequired,
+    tooltipLeftThreshold: PropTypes.number.isRequired,
+    unfocusSlice: PropTypes.func.isRequired,
+    xScale: PropTypes.func.isRequired,
+    yScale: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.willEnter = this.willEnter.bind(this);
+    this.willLeave = this.willLeave.bind(this);
+  }
+
+  getDefaultYPosition(bgBounds, yScale) {
+    const { targetLowerBound, targetUpperBound } = bgBounds;
+    // default Y position is the center of the target range
+    // i.e., 100 mg/dL if target range is 80-120 mg/dL
+    return yScale(targetUpperBound - (targetUpperBound - targetLowerBound) / 2);
```

Comment:
The advice from the react-motion docs is to use instance methods for the `willLeave` and `willEnter` functions, and that makes sense to me given what I've read about React & perf stuff (potential gotcha with things like arrow functions in props because it's a new (update-forcing) function every render cycle). So assuming that stateless components are *not* an option for TransitionMotion-animated components of this kind, does that change your feedback on this `getDefaultYPosition` method? It didn't strike me as particularly problematic, but maybe I'm missing something.

If we derived the value from props and stored it in state we'd need to implement `componentWillReceiveProps` and that seems like premature optimization before we even have the ability to update target range or scales, but OTOH it also seems like a bad React practice to leave it without a `componentWillReceiveProps` (a bug waiting to happen).